web/src/engine/osk/src/keyboard-layout/oskLayer.ts

```diff
@@ -132,42 +136,63 @@ export default class OSKLayer {
       // needing to trigger a reflow by running the closure below early.
       this.spaceBarKey.spec.text = displayName;
 
-      // It sounds redundant, but this dramatically cuts down on browser DOM processing;
-      // but sometimes innerText is reported empty when it actually isn't, so set it
-      // anyway in that case (Safari, iOS 14.4)
-      if (spacebarLabel.innerText != displayName || displayName == '') {
-        spacebarLabel.innerText = displayName;
+      return () => {
+        // It sounds redundant, but this dramatically cuts down on browser DOM processing;
+        // but sometimes innerText is reported empty when it actually isn't, so set it
+        // anyway in that case (Safari, iOS 14.4)
+        if (spacebarLabel.innerText != displayName || displayName == '') {
+          spacebarLabel.innerText = displayName;
+        }
       }
     }
     catch (ex) { }
   }
 
-  public refreshLayout(vkbd: VisualKeyboard, layoutParams: LayerLayoutParams) {
-    const layerHeight = layoutParams.keyboardHeight;
-    this.showLanguage(layoutParams.spacebarText);
-
+  public refreshLayout(layoutParams: LayerLayoutParams) {
```

Comment:
So this implementation of `refreshLayout` doesn't return a closure but other ones do. It would be good if this was clear without having to search for `return` statements.